File: lib/src/interface/rtc_track_event.dart
import 'media_stream.dart';
import 'media_stream_track.dart';
import 'rtc_rtp_receiver.dart';
import 'rtc_rtp_transceiver.dart';

class RTCTrackEvent {
  RTCTrackEvent({
    this.receiver,
    required this.streams,
    required this.track,
    this.transceiver,
  });
  final RTCRtpReceiver? receiver;
  final List<MediaStream> streams;
  final MediaStreamTrack track;
  final RTCRtpTransceiver? transceiver;
}

File: crates/native/src/api/peer/events/rtc_track_event.rs
//! Representation of a [`MediaStreamTrack`] event.

#[cfg(doc)]
use crate::PeerConnection;
use crate::api::{MediaStreamTrack, RtcRtpTransceiver};

/// Representation of a track event, sent when a new [`MediaStreamTrack`] is
/// added to an [`RtcRtpTransceiver`] as part of a [`PeerConnection`].
#[derive(Clone)]
pub struct RtcTrackEvent {
    /// [`MediaStreamTrack`] associated with the [RTCRtpReceiver] identified
    /// by the receiver.
    ///
    /// [RTCRtpReceiver]: https://w3.org/TR/webrtc#dom-rtcrtpreceiver
    pub track: MediaStreamTrack,

    /// [`RtcRtpTransceiver`] object associated with the event.
    pub transceiver: RtcRtpTransceiver,
}
